src/log/test.cc

```diff
@@ -389,6 +389,77 @@ TEST(Log, Speed_nogather)
 }
 
 
+TEST(Log, Multithread)
+{
+  constexpr size_t thread_cnt = 4;
+  std::thread worker[thread_cnt];
+  g_ceph_context->_conf->subsys.set_gather_level(ceph_subsys_context, 30);
+  g_ceph_context->_conf->subsys.set_log_level(ceph_subsys_context, 0);
+
+  auto action = [] (size_t thread_id) -> void {
+    constexpr size_t buf_cnt = 10000;
+    constexpr size_t buf_size = 4096;
+    std::string buffers[buf_cnt];
+    for (size_t i = 0; i < buf_cnt; i++)
+      buffers[i] = std::string(buf_size, (char)rand());
+
+    for (size_t i = 0; i < 200000; i++) {
+      buffers[rand() % buf_cnt] = buffers[rand() % buf_cnt];
```

Comment:
What is this really measuring? I'm not really seeing a trend from https://gist.github.com/aclamk/aeb0faf22c271fddabc81c5efe9fa1e5

Reply:
@batrick The test I made was faulty.
The intention was to measure impact on page tables and tls walks.
The actual result was totally garbled because rand() does protect its context with mutex. This caused all wierd results.

Please ignore this result for now.